extensions/azure/cosmos/assetindex-cosmos/src/main/java/org/eclipse/dataspaceconnector/assetindex/azure/CosmosAssetIndex.java

```diff
@@ -105,6 +106,16 @@ public void accept(Asset asset, DataAddress dataAddress) {
         assetDb.saveItem(assetDocument);
     }
 
+    @Override
+    public Asset deleteById(String assetId) {
+        try {
+            var deletedItem = assetDb.deleteItem(assetId);
+            return convertObject(deletedItem).getWrappedAsset();
+        } catch (NotFoundException notFoundException) {
+            return null;
```

Comment:
log at debug level?

Reply:
I added it. The CosmosAssetIndex did not have a monitor so I added the dependency.